gobblin-yarn/src/main/java/org/apache/gobblin/yarn/GobblinYarnAppLauncher.java

```diff
@@ -342,7 +342,8 @@ public void initializeYarnClients(Config config) {
    * @throws IOException if there's something wrong launching the application
    * @throws YarnException if there's something wrong launching the application
    */
-  public void launch() throws IOException, YarnException, InterruptedException {
+  public void launch()
```

Comment:
should it really be checked?  e.g. how likely is it that a caller would want to handle CNFE specifically?

we might throw something more general, like IOE... or just wrap it in `RuntimeException`. looking at a few other uses of reflection for dynamic instance-construction, I don't see reflection-based failures showing up in the exception signature